configure.ac

```diff
@@ -9,16 +9,29 @@ AC_TYPE_SSIZE_T
 AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 AC_CHECK_FUNCS([memmove pledge reallocarray setlocale strdup])
-AC_SEARCH_LIBS([initscr], [ncursesw ncurses], [], [
-  AC_MSG_ERROR([unable to find initscr() function. Try to install ncurses])
-])
-
-AC_TRY_LINK([
-	#include <ncursesw/curses.h>
-	#include <ncursesw/term.h>
-], [], [
-AC_DEFINE(HAVE_NCURSESW_H,, Define if ncursesw is available)
-])
-
+AC_ARG_WITH([ncursesw],
+  AS_HELP_STRING([--with-ncursesw], [link against ncursesw]),
+  [],
+  [with_ncursesw=no],
+)
+AS_IF([test "x$with_ncursesw" != xno],
+  AC_SEARCH_LIBS([setupterm], [ncursesw],
+    [
+      AC_TRY_LINK(
+        [
+          #include <ncursesw/curses.h>
+          #include <ncursesw/term.h>
+        ],
+        [],
+        [AC_DEFINE([HAVE_NCURSESW_H], [1], [Define if ncursesw is available])],
+        [AC_MSG_ERROR([unable to find ncursesw headers])]
+      )
+    ],
+    [AC_MSG_ERROR([unable to find setupterm function using ncursesw])]
+  ),
+  AC_SEARCH_LIBS([setupterm], [curses], [],
```

Comment:
The old `AC_SEARCH_LIBS` mentioned `ncurses` instead of `curses`. I'm not sure what the differences between them are.